.github/workflows/run_periodic_tests.yml

```diff
@@ -56,46 +101,110 @@ jobs:
         # sometimes gfortran cannot be found, so reinstall gcc just to be sure
         run: |
           brew analytics off
-          brew install graphviz libomp
+          brew install graphviz libomp openblas
           brew reinstall gcc
 
-      - name: Install Windows system dependencies
-        if: matrix.os == 'windows-latest'
-        run: choco install graphviz --version=8.0.5
-
       - name: Set up Python ${{ matrix.python-version }}
+        id: setup-python
         uses: actions/setup-python@v5
         with:
           python-version: ${{ matrix.python-version }}
+          cache: 'pip'
 
-      - name: Install nox
-        run: python -m pip install nox
+      - name: Install nox and tomlkit
+        run: python -m pip install nox tomlkit
 
-      - name: Install SuiteSparse and SUNDIALS on GNU/Linux and macOS
+      - name: Cache pybamm-requires nox environment for macOS
+        uses: actions/cache@v4
+        with:
+          path: |
+            # Headers and dynamic library files for SuiteSparse and SUNDIALS
+            ${{ github.workspace }}/install_KLU_Sundials/
+          key: nox-${{ matrix.os }}-pybamm-requires-${{ steps.setup-python.outputs.python-version }}-${{ hashFiles('**/install_KLU_Sundials.py', '**/noxfile.py', '**/test_on_push.yml') }}
+
+      - name: Install SuiteSparse and SUNDIALS on macOS
         timeout-minutes: 10
-        if: matrix.os != 'windows-latest'
         run: python -m nox -s pybamm-requires
 
       - name: Run unit tests for ${{ matrix.os }} with Python ${{ matrix.python-version }}
-        if: matrix.os != 'ubuntu-latest' || matrix.python-version != '3.12'
         run: python -m nox -s unit
 
-      - name: Run coverage tests for ${{ matrix.os }} with Python ${{ matrix.python-version }}
-        if: matrix.os == 'ubuntu-latest' && matrix.python-version == '3.12'
+      - name: Run integration tests for ${{ matrix.os }} with Python ${{ matrix.python-version }}
+        run: python -m nox -s integration
+
+  run_linux_unit_integration_coverage_tests:
+    needs: style
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version: ["3.9", "3.10", "3.11", "3.12"]
+    name: Tests (ubuntu-latest / Python ${{ matrix.python-version }})
+
+    steps:
+      - name: Check out PyBaMM repository
+        uses: actions/checkout@v4
+
+      - name: Install Linux system dependencies
+        uses: awalsh128/cache-apt-pkgs-action@v1.4.2
+        with:
+          packages: gfortran gcc graphviz pandoc
+          execute_install_scripts: true
```

Comment:
You can replace these with a simple `apt-get install` in the step below, because we don't need to cache anything for scheduled tests.